pycbc/inference/io/cpnest.py

```diff
@@ -27,7 +27,7 @@
 from .posterior import PosteriorFile
 
 
-class CPNestFile(BaseSamplerFile):
+class CPNestFile(BaseSamplerFile, PosteriorFile):
```

Comment:
Don't do this. Keep the inheritance limited to `BaseSamplerFile`. Instead, add a `read_raw_samples` function to this class, and have it call `PosteriorFile.read_raw_samples`, similar to what I did with `write_posterior`.